docs/proposals/dockerfile_component_detection.md

```diff
@@ -0,0 +1,58 @@
+# Enhance port detection for dockerfiles that are not in a component.
+
+## Related GitHub Issues
+* https://github.com/redhat-developer/alizer/issues/235
+
+## Background
+
+What is a component?
+
+Components are defined in the `languages-customization.yml` as a language that has specfic `configuration_files`.
+
+Example of a Go component:
+
+```yaml
+Go:
+  exclude_folders:
+    - "vendor"
+  configuration_files:
+    - "go.mod"
+  component: true
+```
+
+What happens when the repo being scanned has component(s) found but also contains a dockerfile one level outside of the repo?
+
+```shell
+.repo-being-detected-by-alizer
+├── component/
+│   └── dockerfile
+└── dockerfile
+```
+
+The dockerfile in the root level would not be used during analysis because it does not have a component definition in the `languages-customization.yml`
+
+## Potential Solution
+
+Introduce a new component field called `orphan_component` and add it to the `Dockerfile` language:
+
+```yaml
+Dockerfile:
+  configuration_files:
+    - "[Dd]ockerfile"
+  orphan_component: true
+```
+
+and create a component enricher:
+
+```go
+func (d DockerEnricher) DoEnrichComponent(component *model.Component, settings model.DetectionSettings, ctx *context.Context) {
+    // get ports from the dockerfile using our existing dockerfile strategy
+}
+```
+
+Using the `orphan_component` field, we can separate out a dockerfile as a component from what we currently define as a component. This can also be used for any future components that are similar to a dockerfile.
+
+## Potential Problems
+
+* Weak documentation as to what the fields in the `language-customization.yml` currently do.
+* Could have potential naming collision with field names if the `language.yml` file is updated.
```

Comment:
can you elaborate what naming collisions you're anticipating?  I checked the current version of the [linguist languages.yml](https://github.com/github-linguist/linguist/blob/master/lib/linguist/languages.yml#L1540C1-L1554) file and it looks like it has an updated definition for Dockerfile (which includes the Containerfile name).  If we adopt this, would we still have these concerns?

Reply:
The naming collision was more so with introducing fields to the `languages-customization.yml` file. I don't think they would add `orphan_component` as a field, but if they did there would be a collision there so it should be something that we consider